src/libaudqt/audqt.cc

```diff
@@ -34,7 +38,40 @@
 namespace audqt {
 
 static int init_count;
-static QApplication * qapp;
+
+// wrap QApplication only so we can generate a debug trace
+class AudApplication : public QApplication
+{
+public:
+    AudApplication (int &argc, char **argv)
+        : QApplication (argc, argv)
+    {
+        setAttribute (Qt::AA_UseHighDpiPixmaps);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 3, 0)
+        setAttribute (Qt::AA_ForceRasterWidgets);
+#endif
+
+        setApplicationName (_("Audacious"));
+        if (windowIcon().isNull()) {
+            // find the likely icon theme directory
+            QString appIconUrl = QStandardPaths::locate(QStandardPaths::GenericDataLocation, QStringLiteral("icons/hicolor/scalable/apps/audacious.svg"));
+            // load the scalable icon we installed ourselves (QIcon::fromTheme() won't do that)
+            QIcon appIcon = QIcon(appIconUrl);
+            if (appIcon.isNull()) {
+                setWindowIcon (QIcon::fromTheme (argv[0]));
+            } else {
+                setWindowIcon (appIcon);
+            }
```

Comment:
I would like to avoid scattering custom icon-loading code throughout the application.  Let's come up with a common way of handling this.

QtUI already installs its own freedesktop.org-style icon theme, and then sets it up with QIcon::setThemeSearchPaths().  I think using an icon theme is the right way to go.  For one thing, it allows us to take advantage of all of QIcon's handling of different icon sizes, without reinventing the wheel.

However, neither the audacious nor audacious-plugins repositories are the right place to store an icon theme.  My suggestion is to install some theme (user/packager's choice, we don't need a specific "audacious" theme) separately before audacious, then tell audacious where to find it with an argument to ./configure.  audacious can add its own icons (audacious.png/svg) to this theme rather than to hicolor (which I assume isn't part of the standard Mac OS environment).  Then we can use QIcon::setThemeSearchPaths() to point Qt to this theme, and proceed to use QIcon::fromTheme() just as we would on Linux or BSD.